### PR TITLE
implement simple delete for blocks and services

### PR DESCRIPTION
**Pull request by vitiral**

### CODE REVIEW

File: pynio/service.py [hunk 1]
```diff
@@ -75,6 +95,13 @@ def type(self):
     def status(self):
         return self._status().get('status')
 
+    def delete(self):
+        '''Delete self from instance'''
+        self._instance.services.remove(self._name)
+        self._instance._put(
```
Comment (hansmosh) [inline pynio/service.py]: This should be delete

File: pynio/service.py [hunk 2]
```diff
@@ -75,6 +95,13 @@ def type(self):
     def status(self):
         return self._status().get('status')
 
+    def delete(self):
+        '''Delete self from instance'''
+        self._instance.services.remove(self._name)
```
Comment (hansmosh) [inline pynio/service.py]: I think this should only happen if the delete to the rest api is actually successful.

File: pynio/block.py [hunk 3]
```diff
@@ -40,3 +40,11 @@ def name(self, value):
     @property
     def type(self):
         return self._type
+
+    def delete(self):
+        '''Delete self from instance and services'''
+        self._instance.blocks.remove(self._name)
```
Comment (hansmosh) [inline pynio/block.py]: Only remove if the delete rest call is successful.

File: pynio/block.py [hunk 4]
```diff
@@ -40,3 +40,11 @@ def name(self, value):
...
+        for s in self._instance.services.values():
+            s._remove(self)
```
Comment (hansmosh) [inline pynio/block.py]: It looks like this is only removing the block from local configs. Is that necessary? If these modified services are never saved, then nio will still have references to this deleted block. I actually think it's ok to leave the references in the services. What if you are deleting a block named "my_block" and your plan is to create a new "my_block" of a different type but still have it used in all the services?

Comment (cloudformdesign) [inline pynio/block.py]: I think you should do that a different way. I personally think the system of swapping out blocks by name is a terrible convetion and shouldn't be upheld in an api.

If you want to replace a block with a different block, do it explicitly -- not implicitly by using the sane name. If that is something we want, we should implement a `replace` function.

Comment (cloudformdesign) [inline pynio/block.py]: Also, I prefer doing an instance.put -- this way all services + blocks are updated. This does have an issue if there are multiple users though.

Comment (cloudformdesign) [inline pynio/block.py]: Zen #2 "Explicit is better than implicit."

:D

**Pull request merged**